main/model.h

```diff
@@ -6,17 +6,30 @@
 #include <memory>
 #include <vector>
 
-class Model {
+class AbstractModel {
+public:
+  // Model that simulation will be performend on
+  virtual std::vector<objects::TriangleObj *> triangles() const = 0;
+  virtual float height() const = 0;
+  virtual float sideSize() const = 0;
+  virtual bool empty() const = 0;
```

Comment:
When I make empty() method non abstract, buildCollectors wont't build,because model isempty,so I will do: 

`virtual bool empty() const {
  return std::max(height(), sideSize()) <= 0;
}`

Reply:
Why did you revert this back to be abstract?

My question about this method being not clear still holds then.